### PR TITLE
Migrate to new Shipkit plugins

### PR DESCRIPTION
**Pull request by slawekjaranowski**
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [ ] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [ ] The `package-info.java` contains information about the new extension

Code
* [ ] Code adheres to code style, naming conventions etc.
* [ ] Successful tests cover all changes
* [ ] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those 

Checks
* [ ] Tests use [AssertJ](https://joel-costigliola.github.io/assertj/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#assertions) (which are based on AssertJ)
* [x] tested localy by commands:
```
./gradlew  bintrayUpload -PbintrayDryRun
./gradlew generateChangelog
```

Contributing
* [ ] A prepared commit message exists
* [x] The list of contributions inside `README.md` mentions the new contribution (real name optional) 

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

File: version.properties
Comment (beatngu13) [inline version.properties:3]: Does this cover Pioneers versioning scheme (bump minor for feature release, bump patch for hotfix release)?

**Review by beatngu13 — COMMENTED** [review]
Thanks for the PR! 🥳 Just did a quick scan before one of the maintainers finds time.

Afterwards, we shouldn't forget to remove the obsolete secrets (`TRAVIS_BRACN`etc.).

File: version.properties
Comment (mockitoguy) [reply version.properties:3]: Yes. Patch version is incremented automatically with every release. To increment major or minor version one needs to change the `version.properites` file and change `version=1.3.*` to `version=1.4.*` or `version=2.0.*`.

**Comment by mockitoguy** [comment]
@slawekjaranowski, thank you! How did you test this change?

**Comment by slawekjaranowski** [comment]
As I wrote I tested by command:
```
./gradlew  bintrayUpload -PbintrayDryRun
./gradlew generateChangelog
```

File: build.gradle.kts
Comment (slawekjaranowski) [reply build.gradle.kts:27]: it is rather for information, using in `scm.url` next will be displayed , eg on
https://search.maven.org/artifact/org.junit-pioneer/junit-pioneer as as source code
https://bintray.com/junit-pioneer/junit-pioneer/junit-pioneer as vcs

File: .github/workflows/release-build.yml
Comment (slawekjaranowski) [reply .github/workflows/release-build.yml:27]: when job stop print output by sometime (I don't know how long) it will be hang and kill after timeout

Comment (beatngu13) [reply .github/workflows/release-build.yml:27]: Hmmm, do you have a reference for this?

Comment (slawekjaranowski) [reply .github/workflows/release-build.yml:27]: https://github.com/mockito/mockito-testng/issues/26

File: build.gradle.kts
Comment (aepfli) [inline build.gradle.kts:28]: would it make sense to put those properties into the gradle.properties? smae for group and description. To somehow separate the build logic and the hard coded values?

File: build.gradle.kts
Comment (aepfli) [inline build.gradle.kts:275]: this is also part of linke 192 - should we maybe extract it also?

**Review by aepfli — CHANGES_REQUESTED** [review]

File: build.gradle.kts
Comment (slawekjaranowski) [reply build.gradle.kts:28]: `group` and `description` is gradle Project properties, as I see we must set those in `build.gradle.kts`

next I see - when we define something in `gradle.properties` we must refer it in `build.gradle.kts` like 
```
val junitVersion : String by project`
```
so maybe another proposition ...

File: .github/workflows/release-build.yml
Comment (beatngu13) [reply .github/workflows/release-build.yml:27]: Thanks for the link!

These issues seem to be related to Travis CI, not GitHub Actions. For instance, [this issue](https://github.com/bintray/gradle-bintray-plugin/issues/240) or [this suggestion](https://github.com/mockito/mockito-testng/pull/35#issuecomment-762293490). From the [Travis CI docs](https://docs.travis-ci.com/user/common-build-problems/#my-builds-are-timing-out):

>Our builds have a global timeout and a timeout that's based on the output. If no output is received from a build for 10 minutes, it's assumed to have stalled for unknown reasons and is subsequently killed.

I couldn't find anything similar when it comes to GitHub Actions.

Comment (slawekjaranowski) [reply .github/workflows/release-build.yml:27]: It also occurs on GitHub Actions
```
2020-12-07T10:12:08.7002417Z Uploading to https://api.bintray.com/content/mockito/maven/mockito-testng/0.1.5/org/mockito/mockito-testng/0.1.5/mockito-testng-0.1.5-sources.jar...
2020-12-07T10:12:38.6983788Z 
2020-12-07T10:12:38.6985346Z > Task :bintrayPublish
2020-12-07T16:11:35.3475891Z ##[error]The operation was canceled.
```

And with workaround
```
2021-01-18T22:29:46.6942806Z > Task :bintrayPublish
2021-01-18T22:30:40.9698429Z =====[ 120 seconds elapsed -- still running ]=====
2021-01-18T22:31:40.9711465Z =====[ 180 seconds elapsed -- still running ]=====
2021-01-18T22:32:12.2676011Z 
```

Comment (beatngu13) [reply .github/workflows/release-build.yml:27]: Hmmm, isn't this because between:

```
...10:12:38...
...16:11:35...
```

Almost 6 hours elapsed, timeout kicks in. Whereas here:

```
...22:29:46...
...22:32:12...
```

Only a few minutes elapsed.

If the upload takes too long – for whatever reason(s) – it doesn't matter whether the job outputs something. It will be canceled after 360 minutes / 6 hours.

What do you think?

Comment (slawekjaranowski) [reply .github/workflows/release-build.yml:27]: Ok, I will remove it .. we will see if it was need, maybe not

Comment (beatngu13) [reply .github/workflows/release-build.yml:27]: Thank you, let's hope so. 😉

File: build.gradle.kts
Comment (aepfli) [reply build.gradle.kts:28]: yeah i am still not used to the kotlin gradle setup :D i am more aware of the gradle approach.
Still it would be nice to have the actual data defined outside the building blocks, as it is imho easier to maintain. but i might be the only one with this thinking :D - so for me it is fine, not sure what the rest thinks

**Comment by aepfli** [comment]
so we have to rethink this approach anyways, as bintray/jcenter is EOL in may 2021 https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

for now this is fine, but we might have to adapt the process of uploading etc, so we directly push to maven central and sign within gradle

**Review by beatngu13 — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
please re-run GitHub Action for PR

**Comment by Bukama** [comment]
> please re-run GitHub Action for PR

Well that won't restart timeout jobs. Those are only restarted after a new commit.

**Review by aepfli — APPROVED** [review]
LGTM

**Comment by slawekjaranowski** [comment]
Thank you for approval. 
Are there something what I should do before merge?

**Comment by nipafx** [comment]
Thank you so much for the PR, @slawekjaranowski! I'm sorry that I still didn't take a look at it. It's not for a lack or appreciation, I assure you. I hope to get around to it in the coming days.

**Review by Bukama — APPROVED** [review]
LGTM (aside from the merge conflicts)

**Comment by nipafx** [comment]
Thanks again @slawekjaranowski, merged. :)

**Comment by nipafx** [comment]
@slawekjaranowski We just released 1.3.8, but expected 1.3.1 - any idea what happened?

**Comment by slawekjaranowski** [comment]
@nipafx `shipkit-auto-version` plugin is responsible for calculating next version during release. In this case we have new 8 commits from last release so version was incremented by 8.

Please look at https://github.com/shipkit/shipkit-auto-version#implementation-details

**Pull request merged**
